src/main.rs

```diff
@@ -1,30 +1,63 @@
-use std::path::PathBuf;
-
-use axoproject::WorkspaceInfo;
-use camino::Utf8PathBuf;
-
-fn main() -> Result<(), miette::Report> {
-    // Take an optional "path to search from" as the first argument
-    // Otherwise use current working directory
-    let mut args = std::env::args_os();
-    let _bin_name = args.next();
-    let start_path = args.next().map(PathBuf::from);
-
-    let start_dir = start_path
-        .unwrap_or_else(|| std::env::current_dir().expect("couldn't get current working dir!?"));
-    let start_dir = Utf8PathBuf::from_path_buf(start_dir).expect("project path isn't utf8!?");
-
-    if let Some(project) = axoproject::get_project(&start_dir) {
-        print_project(&project);
-    } else {
-        eprintln!("Couldn't find a project in {}", start_dir);
+use axoasset::LocalAsset;
+use axoproject::{WorkspaceInfo, WorkspaceSearch, Workspaces};
+use clap::Parser;
+use miette::Report;
+
+use cli::{OutputFormat, Cli};
+use serde::{Serialize, Deserialize};
+
+mod cli;
+
+fn main() {
+    let config = Cli::parse();
+    axocli::CliAppBuilder::new("axoproject")
+        .verbose(config.verbose)
+        .json_errors(config.output_format == OutputFormat::Json)
+        .start(config, real_main);
+}
+
+fn real_main(app: &axocli::CliApp<Cli>) -> Result<(), Report>  {
+    // FIXME: use the root_dir
+    let root_dir = app.config.root.clone();
+    let start_dir = app.config.search_path.clone().unwrap_or_else(|| LocalAsset::current_dir().unwrap());
+
+    let workspaces = axoproject::get_workspaces(&start_dir);
+    
+    match app.config.output_format {
+        OutputFormat::Human => print_searches(workspaces),
+        OutputFormat::Json => print_searches_json(workspaces),
     }
+
     Ok(())
 }
```

Comment:
i've always felt a little ehhh about the name `real_main` - it tends to make folks more disoriented than oriented. what would you think of something like `app` or `run`? otherwise i'm feeling solid about this interface, the builder is nice! i could imagine us adding the metrics stuff to it as well, which is a great test